### ISSUE TITLE
DETR  VAL  mAP  0.0000

### ISSUE THREAD

**Issue by FPPMXB** [open-mmlab/mmdetection#11262]
MMdetection3.0 Problem: AP values on the DETR validation set are all 0.0000

Conditions:

1. NWPU-VHR-10 data set was selected, including 455 training sets and 195 verification and testing sets.

2. It can be successfully trained on both Faster-rcnn and YOLOv3 models, but during DETR training, the verification index is all 0.0000.

3. Pre-training model is used

Question:

1. All AP values on DETR verification set are 0.0000

2. In addition, it is also a problem when training Co-DETR, that is to say, as long as the DETR based model has this problem.

Is it because the data set is too small? Or do you need to pay special attention to the adjustment of some specific parameters?

**Comment by FPPMXB** [comment]
![QQ图片20231207161207](https://github.com/open-mmlab/mmdetection/assets/127218221/7fc155a3-4fe6-4cfa-a163-044dac50c395)

**Comment by FPPMXB** [comment]
![image](https://github.com/open-mmlab/mmdetection/assets/127218221/8628ddad-b0bf-4339-8860-e2fdbe8c8ecc)

**Comment by kennedyk1** [comment]
I had the same problem in my custom dataset with 1100 images (800 train, 150 test, 150 validation) on DETR, and successfully on YoloX.

![image](https://github.com/open-mmlab/mmdetection/assets/69402765/e6a2f6c7-f42a-4961-b052-3e35590abba0)

**Comment by FPPMXB** [comment]
可能是数据量太小的原因，继续transformer的模型太吃数据了，你可以看csdn这篇文章http://t.csdnimg.cn/hvKdq
。我做了一些验证性的实验做了一些总结，希望对你有帮助



---Original---
From: ***@***.***&gt;
Date: Tue, Dec 12, 2023 23:05 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [open-mmlab/mmdetection] DETR  VAL  mAP  0.0000 (Issue #11262)




 
I had the same problem in my custom dataset with 1100 images (800 train, 150 test, 150 validation) on DETR, and successfully on YoloX.
 
image.png (view on web)

**Comment by neurosynapse** [comment]
Same issue. After doing inference with these checkpoints no detection can be seen. Tried most of the configs available for this model. Tried locally and in cloud. Same results.

**Comment by hysjly** [comment]
When I was training a custom dataset with Mask R-CNN, I encountered the same problem. Has your issue been resolved?（The dataset I used contains 3,000 images.）

**Comment by skywalkerfmc** [comment]
Same  issue

**Comment by sjxxrh** [comment]
Same issue

**Comment by Goddaman** [comment]
same issue, did u fix it?

**Comment by hysjly** [comment]
My&nbsp;issue&nbsp;at&nbsp;the&nbsp;time&nbsp;was&nbsp;that&nbsp;the&nbsp;annotations&nbsp;in&nbsp;the&nbsp;JSON&nbsp;file&nbsp;were&nbsp;incorrect.&nbsp;I&nbsp;hope&nbsp;this&nbsp;is&nbsp;helpful&nbsp;to&nbsp;you.


hysjly
***@***.***








---Original---
From: ***@***.***&gt;
Date: Tue, Nov 11, 2025 22:25 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [open-mmlab/mmdetection] DETR  VAL  mAP  0.0000 (Issue #11262)


Goddaman left a comment (open-mmlab/mmdetection#11262)
 
same issue, did u fix it?
 
&mdash;

**Comment by lumino33** [comment]
In case you guys still need a solution!
I faced the same problem when training with digit detection dataset, and my solutions are:
1. Using pretrained weight by add this to the config file
`load_from = 'https://download.openmmlab.com/mmdetection/v3.0/detr/detr_r50_8xb2-150e_coco/detr_r50_8xb2-150e_coco_20221023_153551-436d03e8.pth' # Load pretrained checkpoint cho detr
`
2. Using larger image size ((1024, 512) instead of (256, 128))
I used (256, 128) because the original mean of image size is about (128, 64), but maybe it's too small so the transformer couldn't learn anything.
